src/lib.rs

```diff
@@ -243,31 +236,34 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_encrypt_decrypt_boxed() {
+    async fn test_encrypt_decrypt_128_gcm() {
         let provider: SimpleKeyProvider<Aes128Gcm> = SimpleKeyProvider::init([1; 16]);
-        let cipher: EnvelopeCipher<Aes128Gcm, Box<dyn KeyProvider<_>>> =
-            EnvelopeCipher::init(Box::new(provider));
-
-        let message = "hello".as_bytes();
+        let cipher: EnvelopeCipher<Aes128Gcm, _> = EnvelopeCipher::init(provider);
+        test_encrypt_decrypt(cipher).await;
 
-        let record = cipher.encrypt(message).await.unwrap();
-        let decrypted = cipher.decrypt(&record).await.unwrap();
+        let provider: SimpleKeyProvider<Aes128Gcm> = SimpleKeyProvider::init([1; 16]);
+        let provider: Box<dyn KeyProvider<_>> = Box::new(provider);
```

Comment:
Is Box needed here?

Reply:
I believe previously, we have tests to ensure that `Box<dyn KeyProvider<S>>` can do the same thing as `SimpleKeyProvider`. I'm just replicating the test here.

We have an implementation of KeyProvider<S> for Box<dyn KeyProvider<S>>, so I think we should keep the test.

Reply:
OK got it. Cool.